Sources/ArcGISToolkit/Components/Offline/OfflineManager.swift

```diff
@@ -73,6 +80,49 @@ class OfflineManager {
             jobCompletionAction?(job)
         }
     }
+    
+    /// Saves the portal item to UserDefaults.
+    /// - Parameters:
+    ///   - id: The portal item ID.
+    ///   - itemJSON: The portal item JSON.
+    func savePortalItem(_ portalItemID: PortalItem.ID, itemJSON: String) {
+        var savedMapIDs = UserDefaults.standard.stringArray(forKey: "offline") ?? []
+        
+        let id = portalItemID.description
+        
+        if !savedMapIDs.contains(id) {
+            savedMapIDs.append(id)
+        }
+        
+        // Save portal item ID.
+        UserDefaults.standard.set(savedMapIDs, forKey: "offline")
+        
+        // Save portal item JSON.
+        UserDefaults.standard.set(itemJSON, forKey: id)
+        
+        loadMaps()
+    }
+    
+    /// Deletes a given portal item from UserDefaults.
+    /// - Parameter portalItemID: The portal item ID.
+    func deletePortalItem(_ portalItemID: PortalItem.ID) {
+        UserDefaults.standard.removeObject(forKey: portalItemID.description)
+        
+        loadMaps()
+    }
+    
+    /// Loads webmap portal items that have been saved to UserDefaults.
+    private func loadMaps() {
+        guard let mapIDs = UserDefaults.standard.array(forKey: "offline") as? [String] else { return }
```

Comment:
The key should be well qualified. See JobManager for the format of the key it uses.